lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -65,7 +65,7 @@ class ListRecordStream extends stream.Transform {
             entries: [{
                 key: changeStreamDocument.documentKey && changeStreamDocument.documentKey._id,
                 type: this._getType(changeStreamDocument.operationType),
-                value: changeStreamDocument.fullDocument && changeStreamDocument.fullDocument.value,
+                value: changeStreamDocument.fullDocument && JSON.stringify(changeStreamDocument.fullDocument.value),
```

Comment:
We already parsed the string before, so i don't see a reason why it can't be re-parsed again. If the object happens to be circular the first try/catch should handle the error

The thing to change here i think is the way we handle those errors, i think returning an empty event is irrelevant as it will be ignored anyway, i changed it to skip the event completely

Reply:
I see. Now my question is: do we really need to parse the same string several time?

Reply:
The thing here is that we can not parse a part of the string. we only need the fullDocument field as a string (this is where the object metadata lies)

Reply:
Oh i just noticed, i said re-parsed when i should have said stringified 😅.
The idea is if it got parsed with no issue then it should stringify with no errors